base/inference.jl

```diff
@@ -3546,7 +3533,7 @@ function invoke_NF(argexprs, etype::ANY, atypes, sv, atype_unlimited::ANY,
                             all = false
                         end
                     end
-                    if UNION_SPLIT_MISMATCH_ERROR && all
```

Comment:
Just to be sure: This was also about wrongly missing a method due to a wrongly empty type intersection?